Dockerfile

```diff
@@ -0,0 +1,203 @@
+        htop \
+        nano \
+        swig \
+        cmake \
+        libboost-all-dev \
+        zlib1g-dev \
+        libbz2-dev \
+        liblzma-dev \
+        locales \
+        pkg-config \
+        libsox-dev
+
+
+
+
+# Install Bazel
+RUN apt-get install -y openjdk-8-jdk
+# Use bazel 0.11.1 cause newer bazel fails to compile TensorFlow (https://github.com/tensorflow/tensorflow/issues/18450#issuecomment-381380000)
+RUN apt-get install -y --no-install-recommends bash-completion g++ zlib1g-dev
+RUN curl -LO "https://github.com/bazelbuild/bazel/releases/download/0.11.1/bazel_0.11.1-linux-x86_64.deb" 
+RUN dpkg -i bazel_*.deb
+
+# Install CUDA CLI Tools
+RUN apt-get install -y cuda-command-line-tools-9-0
+
+# Install pip
+RUN wget https://bootstrap.pypa.io/get-pip.py && \
+    python get-pip.py && \
+    rm get-pip.py
+
+# << END Install base software
+
+
+
+
+# >> START Configure Tensorflow Build
+
+# Clone TensoFlow from Mozilla repo
+RUN git clone https://github.com/mozilla/tensorflow/
+WORKDIR /tensorflow
+RUN git checkout r1.6
+
+
+# GPU Environment Setup
+ENV TF_NEED_CUDA 1
+ENV CUDA_TOOLKIT_PATH /usr/local/cuda
+ENV CUDA_PKG_VERSION 9-0=9.0.176-1
+ENV CUDA_VERSION 9.0.176
+ENV TF_CUDA_VERSION 9.0
+ENV TF_CUDNN_VERSION 7.1.1
+ENV CUDNN_INSTALL_PATH /usr/lib/x86_64-linux-gnu/
+ENV TF_CUDA_COMPUTE_CAPABILITIES 6.0
+
+# Common Environment Setup
+ENV TF_BUILD_CONTAINER_TYPE GPU
+ENV TF_BUILD_OPTIONS OPT
+ENV TF_BUILD_DISABLE_GCP 1
+ENV TF_BUILD_ENABLE_XLA 0
+ENV TF_BUILD_PYTHON_VERSION PYTHON2
+ENV TF_BUILD_IS_OPT OPT
+ENV TF_BUILD_IS_PIP PIP
+
+# Other Parameters
+ENV CC_OPT_FLAGS -mavx -mavx2 -msse4.1 -msse4.2 -mfma
+ENV TF_NEED_GCP 0
+ENV TF_NEED_HDFS 0
+ENV TF_NEED_JEMALLOC 1
+ENV TF_NEED_OPENCL 0
+ENV TF_CUDA_CLANG 0
+ENV TF_NEED_MKL 0
+ENV TF_ENABLE_XLA 0
+ENV PYTHON_BIN_PATH /usr/bin/python2.7
+ENV PYTHON_LIB_PATH /usr/lib/python2.7/dist-packages
+
+# << END Configure Tensorflow Build
+
+
+
+
+# >> START Configure Bazel
+
+# Running bazel inside a `docker build` command causes trouble, cf:
+#   https://github.com/bazelbuild/bazel/issues/134
+# The easiest solution is to set up a bazelrc file forcing --batch.
+RUN echo "startup --batch" >>/etc/bazel.bazelrc
+# Similarly, we need to workaround sandboxing issues:
+#   https://github.com/bazelbuild/bazel/issues/418
+RUN echo "build --spawn_strategy=standalone --genrule_strategy=standalone" \
+    >>/etc/bazel.bazelrc
+
+# Put cuda libraries to where they are expected to be
+RUN ln -s /usr/local/cuda/lib64/stubs/libcuda.so /usr/local/cuda/lib64/stubs/libcuda.so.1
+RUN cp /usr/include/cudnn.h /usr/local/cuda/include/cudnn.h
+
+# Set library paths
+ENV LD_LIBRARY_PATH $LD_LIBRARY_PATH:/usr/local/cuda/extras/CUPTI/lib64:/usr/local/cuda/lib64:/usr/lib/x86_64-linux-gnu/:/usr/local/cuda/lib64/stubs/
+
+# << END Configure Bazel
+
+
+
+
+# Copy DeepSpeech repo contents to container's /DeepSpeech
+COPY . /DeepSpeech/
+
+WORKDIR /DeepSpeech
+
+RUN pip --no-cache-dir install -r requirements.txt
```

Comment:
Why would you need to do that ? It's only needed for running training

Reply:
Cause I made it to train)

Reply:
Oh, so all my comments after makes no sense :-). I might be picky, but I would have wrote two different Dockerfile, one taking care of building pieces, the other one gathering everything for training / inference

Reply:
Yeah, 2 Dockerfiles - makes sense. I was thinking only about experimenting part)